cucumber-selenium-tests/src/test/java/com/skillstorm/StepDefinitions/SDTransactions.java

```diff
@@ -0,0 +1,139 @@
+package com.skillstorm.StepDefinitions;
+
+
+import org.openqa.selenium.support.ui.WebDriverWait;
+import org.testng.Assert;
+
+import com.skillstorm.WebDriverSingleton;
+import com.skillstorm.PageObjects.LoginPage;
+import com.skillstorm.PageObjects.TransactionPage;
+import com.skillstorm.Utilities.Navigator;
+import com.skillstorm.Utilities.UserData.User;
+import com.skillstorm.Utilities.UserData.UserType;
+
+import java.time.Duration;
+
+import org.openqa.selenium.WebDriver;
+import io.cucumber.java.After;
+import io.cucumber.java.Before;
+import io.cucumber.java.en.*;
+
+public class SDTransactions {
+
+    WebDriver driver;
+    WebDriverWait wait;
+    Navigator navigator;
+    private TransactionPage transactions;
+    private String[] transactionContent;
+
+    /**
+     * Set up the ChromeDriver and initialize TransactionPage.
+     */
+    @Before("@transaction")
+    public void setUp() {
+        driver = WebDriverSingleton.getDriver();
+        wait = new WebDriverWait(driver, Duration.ofSeconds(10));
+        navigator = new Navigator(driver);
+
+        this.transactions = new TransactionPage(driver);
+
+        transactionContent =  new String[4];
+    }
+
+    /**
+     * Tear down the WebDriver after each test.
+     */
+    @After("@transaction")
+    public void tearDown() {
+        WebDriverSingleton.quitDriver();
+    }
+
+    /**
+     * Login Scenario Definitions
+     */
+    @Given("I login")
+    public void login() {
+        navigator.navigateTo(Navigator.PGNAME_LOGIN);
+        LoginPage loginPage = new LoginPage(driver);
+        User user = new User(UserType.PERSISTANT, "joseph.sam@gmail.com", "password1");
+        loginPage.login(user);
+    }
+
+    /**
+     * General Scenario Definitions
+     */
+    @Given("I am on the Transactions page")
+    public void iAmOnTheTranactionsPage() {
+        this.transactions.clickTab();
+        String expectedUrl = "http://localhost:5173/dashboard/transactions";
```

Comment:
High impact: During pipeline integration this would automatically fail inside of a dev environment.
Suggestion: Use the Navigator.PGNAME_TRANSACTIONS as that will pull the URL depending on the environment.